### PR TITLE
Add OSDe2e test harness scaffolding

### PR DESCRIPTION
**Pull request by YiqinZhang**
- Add OSDe2e test harness scaffolding with empty test cases. This will build the pipeline and help migrate tests from OSDe2e in the next step.
- Add dockerfile and make targets for osde2e test harness without boilerplate

[SDCICD-1395](https://issues.redhat.com//browse/SDCICD-1395)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -186,3 +186,28 @@ docs:
...
+######################
+
+# create e2e scaffolding
+
+.PHONY: e2e-harness-generate
+e2e-harness-generate:
+	$ osde2e/e2e-harness-generate.sh $(OPERATOR_NAME) 
+
+# create binary
+.PHONY: e2e-harness-build
```
[the remark this answers is not in the document]
Comment (YiqinZhang) [reply Makefile:201]: Yes, I’ve successfully tested and built the harness test locally.

File: harness.test
Comment (tnierman) [inline harness.test:1]: Looks like this is a binary file that might've accidentally been added to this commit - can we remove it before merging?

File: osde2e/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# THIS FILE IS GENERATED BY BOILERPLATE. DO NOT EDIT.
```
Comment (jbpratt) [inline osde2e/Dockerfile:1]: Please update, this file is not generated by boilerplate

File: osde2e/test-harness-template.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# THIS FILE IS GENERATED BY BOILERPLATE. DO NOT EDIT.
```
Comment (jbpratt) [inline osde2e/test-harness-template.yml:1]: this needs be updated

File: Makefile [hunk 4]
```diff
@@ -186,3 +187,28 @@ docs:
...
+# Use current commit as harness image tag
+CURRENT_COMMIT=$(shell git rev-parse --short=7 HEAD)
+HARNESS_IMAGE_TAG=$(CURRENT_COMMIT)
+
+.PHONY: e2e-image-build-push
+e2e-image-build-push:
+	${CONTAINER_ENGINE} build --pull -f osde2e/Dockerfile -t $(HARNESS_IMAGE_REGISTRY)/$(HARNESS_IMAGE_REPOSITORY)/$(HARNESS_IMAGE_NAME):$(HARNESS_IMAGE_TAG) .
+	${CONTAINER_ENGINE} tag $(HARNESS_IMAGE_REGISTRY)/$(HARNESS_IMAGE_REPOSITORY)/$(HARNESS_IMAGE_NAME):$(HARNESS_IMAGE_TAG) $(HARNESS_IMAGE_REGISTRY)/$(HARNESS_IMAGE_REPOSITORY)/$(HARNESS_IMAGE_NAME):latest
+	${CONTAINER_ENGINE} push $(HARNESS_IMAGE_REGISTRY)/$(HARNESS_IMAGE_REPOSITORY)/$(HARNESS_IMAGE_NAME):$(HARNESS_IMAGE_TAG)
+	${CONTAINER_ENGINE} push $(HARNESS_IMAGE_REGISTRY)/$(HARNESS_IMAGE_REPOSITORY)/$(HARNESS_IMAGE_NAME):latest
```
Comment (jbpratt) [inline Makefile:214]: I'd put these in the makefile in osde2e along with the variables, no need to put them here and include another one as well
